app/views/posts/show.html.haml

```diff
@@ -27,8 +30,12 @@
 .booyah-box.col-xs-10.col-xs-offset-1
   %h1 Comments
   - @post.comments.each do |comment|
-    %div
-      %blockquote
-        \-
-        \#{comment.message}
-        %small= comment.user_id
\ No newline at end of file
+    %blockquote
+      \#{comment.message}
+      %small= comment.user_id
+      - if comment.controlled_by?(current_user)
+        = link_to 'Destroy', comment_path(comment), :method => :delete, :class => 'btn btn-xs btn-primary pull-right'
+        = link_to 'Edit', edit_comment_path(comment), :method => :edit, :class => 'btn btn-xs btn-primary pull-right'
```

Comment:
this line is wrong.  Most notably this part.

```
:method => :edit
```

Links are `GET`s, by default, which is what we want.  There is no HTTP status code `edit`, presumably rails is interpreting random methods as `POST`s. I'd remove that little snippet.